controllers/vspherecluster_reconciler.go

```diff
@@ -48,6 +48,8 @@ import (
 	infrautilv1 "sigs.k8s.io/cluster-api-provider-vsphere/pkg/util"
 )
 
+const legacyIdentityFinalizer string = "identity/infrastructure.cluster.x-k8s.io"
```

Comment:
Do add an explicit comment marking it as Deprecated, kept only for upgrade issues.